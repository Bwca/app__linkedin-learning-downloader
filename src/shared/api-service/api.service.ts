import axios, { AxiosResponse } from 'axios';

class ApiService {
  private readonly httpClient = axios;

  public get<T, K = void, H = void>(url: string, params?: K, headers?: H): Promise<AxiosResponse<T>> {
    return this.httpClient.get<T>(url, { params, headers });
  }
}

export const apiService = new ApiService();
