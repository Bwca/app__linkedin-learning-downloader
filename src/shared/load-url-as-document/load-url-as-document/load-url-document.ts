import { JSDOM } from 'jsdom';
import { apiService } from '../../api-service';

export async function loadUrlAsDocument(url: string): Promise<Document> {
  const { data: html } = await apiService.get<string>(url);

  const {
    window: { document },
  } = new JSDOM(html);

  return document;
}
