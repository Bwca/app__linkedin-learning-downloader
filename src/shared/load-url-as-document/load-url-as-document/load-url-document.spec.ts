import { AxiosResponse } from 'axios';

import { apiService } from "../../api-service";
import { loadUrlAsDocument } from './load-url-document';

describe('loadCoursesDocument', () => {
  it('should return js Document from url', async () => {
    // Arrange
    const url = 'http://something.com';
    jest.spyOn(apiService, 'get').mockResolvedValueOnce({ data: mockHtml() } as AxiosResponse);

    // Act
    const doc = await loadUrlAsDocument(url);

    // Assert
    expect(doc).toBeTruthy();
    expect(doc).toMatchSnapshot();
  });
});

function mockHtml(): string {
  return `<!DOCTYPE html>
    <html>
    <body>
    
    <h1>My First Heading</h1>
    
    <p>My first paragraph.</p>
    
    </body>
    </html>
    `;
}
