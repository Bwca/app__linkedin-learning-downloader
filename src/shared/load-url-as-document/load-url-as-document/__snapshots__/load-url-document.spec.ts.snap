// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadCoursesDocument should return js Document from url 1`] = `
Document {
  "location": Location {
    "assign": [Function],
    "hash": "",
    "host": "",
    "hostname": "",
    "href": "about:blank",
    "origin": "null",
    "pathname": "blank",
    "port": "",
    "protocol": "about:",
    "reload": [Function],
    "replace": [Function],
    "search": "",
    "toString": [Function],
  },
}
`;
